compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -1341,8 +1341,9 @@ impl<'a> Parser<'a> {
                     .span_to_snippet(pat.span.trim_start(begin_par_sp).unwrap())
                     .unwrap_or_else(|_| pprust::pat_to_string(&pat));
 
-                self.struct_span_err(self.prev_token.span, "unexpected closing `)`")
-                    .span_label(begin_par_sp, "opening `(`")
+                let sp = MultiSpan::from_spans(vec![begin_par_sp, self.prev_token.span]);
+
+                self.struct_span_err(sp, "unexpected parenthesis surrounding `for` loop head")
```

Comment:
We should use `multipart_suggestion` here and use the same spans we use for the error.